src/olympia/amo/management/commands/initialize_data.py

```diff
@@ -0,0 +1,201 @@
+
+    def create_db(self):
+        logging.info('Cleaning up directories linked to database records...')
+        root = os.path.join('/', 'data', 'olympia')
+        clean_dirs = (
+            os.path.join(root, 'user-media'),
+            os.path.join(root, 'tmp'),
+        )
+
+        for dir in clean_dirs:
+            if os.path.exists(dir):
+                logging.info(f'Cleaning up {dir}...')
+                os.rmdir(dir)
+
+        database_name = self.db_info.get('NAME')
+        character_set = self.db_info.get('OPTIONS').get('charset', 'utf8mb4')
+
+        if self.db_exists:
+            drop_query = f'DROP DATABASE `{database_name}`'
+            logging.info('Executing... "' + drop_query + '"')
+            self.connection.query(drop_query)
+
+        create_query = (
+            f'CREATE DATABASE `{database_name}` CHARACTER SET {character_set}'
+        )
+        logging.info('Executing... "' + create_query + '"')
+        self.connection.query(create_query)
+
+    def seed_db(self):
+        logging.info('Creating seed data...')
+        # reindex --wipe will force the ES mapping to be re-installed. Useful to
+        # make sure the mapping is correct before adding a bunch of add-ons.
+        call_command('reindex', '--wipe', '--force', '--noinput')
+        call_command('generate_addons', '--app', 'firefox', self.num_addons)
+        call_command('generate_addons', '--app', 'android', self.num_addons)
+        call_command('generate_themes', self.num_themes)
+        # These add-ons are specifically useful for the addons-frontend
+        # homepage. You may have to re-run this, in case the data there
+        # changes.
+        call_command('generate_default_addons_for_frontend')
+
+    def load_initial_data(self):
+        logging.info('Loading initial data...')
+        call_command('loaddata', 'initial.json')
+        call_command('import_prod_versions')
+        call_command(
+            'createsuperuser',
+            '--no-input',
+            '--username',
+            'local_admin',
+            '--email',
+            'local_admin@mozilla.com',
+        )
+        call_command('loaddata', 'zadmin/users')
+
+    def handle(self, *args, **options):
+        """
+        Create the database.
+        """
+        force_db = options.get('force_db')
+        skip_seed = options.get('skip_seed')
+        skip_index = options.get('skip_index')
+
+        logging.info(f'options: {options}')
+
+        # Initialize ES inside the handle method
+        ES = get_es()
+
+        # Step 1: Ensure the database exists
+        # is migrated and contains initial data if creating.
+        create_new_db = force_db or not self.db_exists
+
+        # only create the db if we want to or need to
+        self.create_db() if create_new_db else logging.info('Database already exists.')
+
+        # Migrate database even if not creating anew.
+        logging.info('Migrating...')
+        call_command('migrate', '--noinput')
```

Comment:
What happens if there is a clashing data migration, or some other db incompatibility, at this point?  And what would be the recovery - e.g. do we need to call `make up` again after whatever tinkering is complete? do we suggest forcing a new database?
(should we document this)?

Reply:
I can add to the docs how to recover. That's a good point. Point yes, if a migration fails make up will fail and the solution is generally to run makeup after fixing the issue. 

I see this as a feature meaning that you shouldn't be able to run the app with broken migrations.

Reply:
You may need to be able to run the app - at least `make dbshell` - to fix the problem.

Reply:
If you look closely, the initialize_data command is ran only after the compose up is ran. So you'd be able to do this.